sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/identifiers.scala

```diff
@@ -37,16 +41,44 @@ sealed trait IdentifierWithDatabase {
     val replacedId = quoteIdentifier(identifier)
     val replacedDb = database.map(quoteIdentifier(_))
```

Comment:
shall we add a `replacedCatalog`?

Reply:
I assume the catalog in this v1 identifier will always be spark_catalog.. do you mean it can be changed ?

Reply:
oh, you are right

Reply:
this doesn't matter, but I think it's still better to do it to keep consistent with table and database name, otherwise we need to explain to people that the catalog name is always `spark_catalog` so we don't need to quote it.

Reply:
unified them